src/App.tsx

```diff
@@ -1,9 +1,27 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [movies, setNewMovies] = useState<Movie[]>(moviesFromServer);
+
+  const handlerQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  useEffect(() => {
+    setNewMovies(moviesFromServer.filter(movie => {
+      const movieTitle = movie.title.toLowerCase();
+      const movieDiscription = movie.description.toLowerCase();
+      const movieQuery = query.toLowerCase();
```

Comment:
movieQuery - the same on all iterations you can move it outside filtering function